### DISCUSSION TITLE
Tuning of the avoid roadblock procedure

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#345] [Q&A] [answered]
Don't use "avoid roadblock" often, because I follow the deviation signs or try to find back to the route on my own. 

But recently i had the following situation: I'm standing at the traffic light of an intersection and wait for green. I see that the route I want to take is blocked and initiate "avoid roadblock". This will recommend me an u-turn (even that isn't possible there) and the red route, because the blockage is set equal to my position. Btw: I have no chance to initiate "avoid roadblock" directly in the intersection.

If there is an offset (e. g. 200m - from my position to the roadblock), the green route would be possible. 

<img width="300" src="https://github.com/devemux86/cruiser/assets/121806238/c8d41b67-e488-47d7-b702-eb2face64427"/> 

I remembered that a similar issue has already been raised at K1: https://forum.kurviger.de/t/improved-road-block-handling/2935

Just an (old) idea ...

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for the suggestion.

Can you post the intersection coordinates for tests?
Did you use BRouter (offline)?

This is an interesting example and we can think how we can improve the avoid roadblock function.

**Reply by WolfgangZ42** [reply]
Sure - this was my position: 48.16786,11.565363 
and yes - BRouter (offline)

**Comment by devemux86** [comment]
We could make the avoid roadblock to start a distance ahead of user's location.
Fixed 50-100 m? Adjusted in settings?

**Reply by WolfgangZ42** [reply]
100m fix should work in most cases, I guess. Not necessary to spend another parameter in settings for it.

**Comment by devemux86** [comment]
> We could make the avoid roadblock to start a distance ahead of user's location.

It will be available (100 m) in the next Cruiser update.

<img src="https://github.com/devemux86/cruiser/assets/3484020/25881e8f-9094-4d66-8055-0ae0743f022a" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/e5f8b9b5-8b48-4003-9548-bd3677d0ad97" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/d2332c73-bda9-45b5-a103-63dc3fd8fae6" width="33%"/>

**Comment by devemux86** [comment, marked as answer]
> We could make the avoid roadblock to start a distance ahead of user's location.

Available (100 m) in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
